packages/compass/src/main/application.js

```diff
@@ -193,7 +193,7 @@ Application.prototype.setupLogging = function() {
       ? path.join(appData || home, 'mongodb', 'compass')
       : path.join(home, '.mongodb', 'compass');
   app.setAppLogsPath(logDir);
-  setupLogging();
+  setupLogging().then(logFilePath => { this.logFilePath = logFilePath; });
```

Comment:
Isn't it a bit dodgy that this happens asynchronously? Any way we could wait here or is this ultimately because this gets called in the Application constructor? I suppose it is super unlikely that someone would click Open Log File before this.logFilePath is set and maybe I'm just paranoid.

Reply:
The line that starts spawning windows is `require('./window-manager')` a few lines above, so if you want to, you probably can refactor this file a bit and make this application startup an async process and prevent electron window showing before logs are setup